File: briefly/modules/briefly-transcriber/ios/BrieflyOnDeviceSummarizer.swift
import Foundation

enum BrieflyOnDeviceSummarizer {
  static func summarize(text: String) -> [String: Any] {
    let sentences = text
      .components(separatedBy: CharacterSet(charactersIn: ".!?"))
      .map { $0.trimmingCharacters(in: .whitespacesAndNewlines) }
      .filter { $0.count > 20 }

    let summary = sentences.prefix(3).joined(separator: ". ")
    let keywords = ["decide", "action", "will", "should", "must", "key", "important"]
    let insights = sentences
      .filter { s in keywords.contains { s.lowercased().contains($0) } }
      .prefix(5)
      .map { $0 }

    return [
      "summary": summary.isEmpty ? String(text.prefix(200)) : summary,
      "keyInsights": insights.isEmpty && !sentences.isEmpty ? [sentences[0]] : Array(insights),
    ]
  }
}

File: briefly/modules/briefly-transcriber/android/src/main/java/expo/modules/brieflytranscriber/BrieflyOnDeviceSummarizer.kt
package expo.modules.brieflytranscriber

object BrieflyOnDeviceSummarizer {
  fun summarize(text: String): Map<String, Any> {
    val sentences = text
      .split(Regex("[.!?]+"))
      .map { it.trim() }
      .filter { it.length > 20 }

    val summary = sentences.take(3).joinToString(". ")
    val keywords = listOf("decide", "action", "will", "should", "must", "key", "important")
    val insights = sentences
      .filter { sentence -> keywords.any { keyword -> sentence.lowercase().contains(keyword) } }
      .take(5)

    val keyInsights: List<String> = when {
      insights.isNotEmpty() -> insights
      sentences.isNotEmpty() -> listOf(sentences[0])
      else -> emptyList()
    }

    return mapOf(
      "summary" to summary.ifEmpty { text.take(200) },
      "keyInsights" to keyInsights,
    )
  }
}
